src/database/gravity-db.c

```diff
@@ -302,9 +310,13 @@ static bool domain_in_list(const char *domain, sqlite3_stmt* stmt)
 	// all host parameters to NULL.
 	sqlite3_clear_bindings(stmt);
 
-	// Return result.
+	// Return early in case we already have found an exact match.
 	// SELECT EXISTS(...) either returns 0 (false) or 1 (true).
-	return result == 1;
+	if(result == 1)
+		return true;
+
+	// If not: Walk regex-based whitelist filters in addition
+	return match_regex(domain, REGEX_WHITELIST);
```

Comment:
This function is executed for both whitelist and audit list, so the whitelist-specific code should be lifted into `in_whitelist`.